src/analyst.py

```diff
@@ -583,4 +583,131 @@ def summarize_stats(self) -> dict:
+
+        degree = {}
+        country = {}
+        state = {}
+        education = {}
+
+        for respondent in student_list:
+            utils.update_dict_counter(degree, respondent.student['degree'])
+            utils.update_dict_counter(country, respondent.student['country'])
+            utils.update_dict_counter(state, respondent.student['state'])
+            utils.update_dict_counter(education, respondent.student['education'])
+        
+        res = dict()
+        res['degree']           = degree
+        res['country']          = country
+        res['state']            = state
+        res['education']        = education
+
+        return res
+
+    def summarize_student_ideal(self, respondents_list=None) -> dict:
+        """
+        Summarize students' ideal career for those who completed the "Student" questions
+        """
+
+        if respondents_list is None:
+            respondents_list = self.respondents_list
+
+        student_list = self.filter_respondents_on(is_student=True,is_completed_all_questions=True)
+
+        # needs LLM for job title
+        # ideal_job_title_list = []
+
+        # building other summary for now
+        ideal_vc_counter = dict()
+        ideal_job_aspects_counter = dict()
+        ideal_salary_list = []
+
+        for respondent in student_list:
+            # utils.nanappend(ideal_job_title_list, respondent.student['ideal_job_title'])
+            utils.update_dict_counter(ideal_vc_counter, respondent.student['ideal_value_chain'])
+            utils.update_dict_counter(ideal_job_aspects_counter, respondent.student['ideal_job_aspects'])
+            utils.nanappend(ideal_salary_list, respondent.student['ideal_salary'])
+
+        res = dict()
+        # res['ideal_job_title_list'] = ideal_job_title_list
+        res['ideal_value_chain'] = ideal_vc_counter
+        res['ideal_job_aspects'] = ideal_job_aspects_counter
+        res['ideal_salary_list'] = np.array(ideal_salary_list)
+
+        # these numbers aren't very good... may need to manually pick out outliers
+        res['ideal_salary_median'] = np.nanmedian(ideal_salary_list)
+        res['ideal_salary_mean'] = np.nanmean(ideal_salary_list)
+        res['ideal_salary_std'] = np.nanstd(ideal_salary_list)
+
+        return res
+
+    def summarize_student_internship(self, respondents_list=None) -> dict:
+        """
+        Summarize students' internship experience for those who completed the "Student" questions
+        """
+
+        if respondents_list is None:
+            respondents_list = self.respondents_list
+
+        student_list = self.filter_respondents_on(is_student=True,is_completed_all_questions=True)
+
+        num_internships_counter = dict()
+        internship_vc_counter = dict()
+        internship_role_counter = dict()
+        internship_hourly_pay_list = []
+        internship_hours_per_week_list = []
+        # to be implemented with LLM: vc, role, skills
```

Comment:
Please make sure `Analyst` returns the list of strings for each open ended text field. Analyst is going to aggregate the list of strings for the LLM to handle. Analyst shouldn't know anything about how the LLM works and vice versa (separation of responsibilities).